tests/robots/test_robot.py

```diff
@@ -62,5 +68,93 @@ def test_basic_name_links(ur5_links):
 
 
 def test_basic_attr():
-    robot = Robot.basic('testbot', location="rfl")
-    assert robot.model.attr['location'] == "rfl"
+    robot = Robot.basic('testbot', location='rfl')
+    assert robot.model.attr['location'] == 'rfl'
+
+
+def test_name(panda_robot_instance):
+    robot = panda_robot_instance
+    assert robot.name == 'panda'
+
+
+def test_group_names(ur5_robot_instance):
+    robot = ur5_robot_instance
+    assert robot.group_names == ['manipulator', 'endeffector']
+
+
+def test_main_group_name(panda_robot_instance):
+    robot = panda_robot_instance
+    assert robot.main_group_name == 'panda_arm_hand'
+
+
+def test_root_name(ur5_robot_instance):
+    robot = ur5_robot_instance
+    assert robot.root_name == 'world'
+
+
+@pytest.mark.parametrize('group,expected_result', [(None, 'panda_rightfinger'), ('panda_arm_hand', 'panda_rightfinger'), ('panda_arm', 'panda_link8')])
+def test_get_end_effector_link_name(panda_robot_instance, group,  expected_result):
+    robot = panda_robot_instance
+    assert robot.get_end_effector_link_name(group=group) == expected_result
+
+
+def test_get_end_effector_link_name_wrong_group(panda_robot_instance):
+    robot = panda_robot_instance
+    with pytest.raises(KeyError):
+        robot.get_end_effector_link_name(group='panda_leg')
+
+
+@pytest.mark.parametrize('group', [None, 'endeffector'])
+def test_get_end_effector_link(ur5_robot_instance, group):
+    robot = ur5_robot_instance
+    assert isinstance(robot.get_end_effector_link(group=group), compas.robots.Link)
```

Comment:
Again, much simpler to read without the parametrization:
```suggestion
def test_get_end_effector_link(ur5_robot_instance):
    robot = ur5_robot_instance
    assert isinstance(robot.get_end_effector_link(group=None), compas.robots.Link)
    assert isinstance(robot.get_end_effector_link(group='endeffector'), compas.robots.Link)
```

I would also argue this test is not very useful. Instead of testing for `isinstance`, you could -for example- check that the name of the end effect is the one you expect.